### ISSUE TITLE
battery percentage gets set to gibberish once it is updated in usettings 

### ISSUE THREAD

**Issue by worthless443** [worthless443/csysteminfo#1]
when the battery percentage goes from `x` to `x+1`, `_atoi()`or `atoi()` can't find numbers to convert to and set to `0`. It is also the case when the percentage is <10.

**Comment by worthless443** [comment]
running `./test` for few increments
```
 ERROR (1/3)
 ERROR (2/78)
 ERROR (3/276)
 ERROR (4/396)
 ERROR (5/397)
 ERROR (6/573)
 ERROR (7/581)
 ERROR (8/596
```

**Comment by worthless443** [comment]
it seems like when battery percentage is at 100%, there is some parsing bug resulting in segmentation or other memory faults.

**Comment by worthless443** [comment]
fixed in 889ecc0dee97e79d653481b406284ee675b354cf
